aspnetcore/fundamentals/routing.md

```diff
@@ -285,168 +394,169 @@ The methods provided by <xref:Microsoft.AspNetCore.Routing.LinkGenerator> suppor
+This is a summary of the steps that routing performs with the template `/a{b}c{d}` and the URL path `/abcd`. The `|` is used to help visualize how the algorithm works:
 
-This is a summary of the steps that routing would perform with the template `/a{b}c{d}` and the URL path `/abcd`. The `|` is used to help visualize how the algorithm works.
-* The first literal (right to left) is `c`, so `/abcd` is searched from right and found `/ab|c|d`.
+* The first literal, right to left, is `c`. So `/abcd` is searched from right and finds `/ab|c|d`.
 * Everything to the right (`d`) is now matched to the route parameter `{d}`.
-* The next literal (right to left) is `a`, so `/ab|c|d` is searched starting where we left off, then `a` is found `/|a|b|c|d`. 
+* The next literal, right to left, is `a`. So `/ab|c|d` is searched starting where we left off, then `a` is found `/|a|b|c|d`.
 * The value to the right (`b`) is now matched to the route parameter `{b}`.
-* There is no remaining text and no remaining route template so this is a match.
+* There is no remaining text and no remaining route template, so this is a match.
 
-Now here's an example of a negative case using the same template `/a{b}c{d}` and the URL path `/aabcd`. The `|` is used to help visualize how the algorithm works. This case *is not* a match, which is explained by the same algorithm.
-* The first literal (right to left) is `c`, so `/aabcd` is searched from right and found `/aab|c|d`.
+Here's an example of a negative case using the same template `/a{b}c{d}` and the URL path `/aabcd`. The `|` is used to help visualize how the algorithm works. This case isn't a match, which is explained by the same algorithm:
+* The first literal, right to left, is `c`. So `/aabcd` is searched from right and finds `/aab|c|d`.
 * Everything to the right (`d`) is now matched to the route parameter `{d}`.
-* The next literal (right to left) is `a`, so `/aab|c|d` is searched starting where we left off, then `a` is found `/a|a|b|c|d`. 
+* The next literal, right to left, is `a`. So `/aab|c|d` is searched starting where we left off, then `a` is found `/a|a|b|c|d`.
 * The value to the right (`b`) is now matched to the route parameter `{b}`.
 * At this point there is remaining text `a`, but the algorithm has run out of route template to parse, so this is not a match.
 
-Since the matching algorithm is non-greedy - it matches the smallest amount of text possible in each step - any case where the *delimiter value* will appear inside the parameter values will result in no match. Regular expressions provide much more control over their matching behavior.
+Since the matching algorithm is [non-greedy](#greedy):
+
+* It matches the smallest amount of text possible in each step.
+* Any case where the delimiter value appears inside the parameter values results in not matching.
+
+Regular expressions provide much more control over their matching behavior.
+
+<a name="greedy"></a>
+
+#### Greedy route matching
```

Comment:
I'm not sure we we have a heading just for this definition, this just explains some of the terminology used in the complex segment discussion, and it doesn't really apply to routing (we don't have features that do greedy matching).